File: Sources/NSArray-Extension.m
#import "NSArray-Extension.h"

@implementation NSArray (Extension)
- (NSIndexSet*)indexesOfTrueValue
{
    return [self indexesOfObjectsPassingTest:^BOOL(NSNumber *obj, NSUInteger idx, BOOL *stop) {
        return (obj.boolValue == YES);
    }];
}

- (NSArray*)mapUsingBlock:(id (^)(id))block
{
    NSMutableArray *array = [NSMutableArray array];
    [self enumerateObjectsUsingBlock:^(id item, NSUInteger idx, BOOL *stop) {
        [array addObject:block(item)];
    }];
    return array;
}
@end

File: Sources/NSArray-Extension.swift
import Foundation

@objc extension NSArray {
    @objc func indexesOfTrueValue() -> IndexSet {
        indexesOfObjects { obj, _, _ in
            (obj as? NSNumber)?.boolValue == true
        }
    }

    @objc func mapUsingBlock(_ block: @escaping (Any) -> Any) -> NSArray {
        var result = [Any]()
        result.reserveCapacity(count)
        for item in self {
            result.append(block(item))
        }
        return result as NSArray
    }
}